.gitignore

```diff
@@ -147,3 +147,9 @@ build
 
 # Vscode
 .vscode/
+
+*.iml
+java/test
+java/*/target
+java/target/
```

Comment:
Don't ignore `java/test`, it's a valid module.
`java/test` and `java/*/target` can be combined as `java/**/target`